BlueBrain/Search
# [BBS-213] Add Save/Load buttons in notebook

# What's New?

- This PR introduces the buttons "Save" and "Load" for both the Search and the Mining Widget.

<img width="400" alt="Screenshot 2020-12-03 at 18 01 57" src="https://user-images.githubusercontent.com/17013890/101062334-cebdb000-3591-11eb-88ca-77ad1249a291.png">
<img width="400" alt="Screenshot 2020-12-03 at 18 01 46" src="https://user-images.githubusercontent.com/17013890/101062410-e7c66100-3591-11eb-9c9e-1ff6eb813ff1.png">

- These buttons allow to save the results of a search / mining action, so that in case of OOD session timeout, kernel dead, ... we can quickly recover from the point where we left without having to click on "Search Literature!" and/or "Mine Articles!" and having to wait for a potentially long time!
<img width="400" alt="Screenshot 2020-12-03 at 18 12 53" src="https://user-images.githubusercontent.com/17013890/101063527-2f012180-3593-11eb-90d6-a3c5dc53a960.png">

- **Note**: the functionality implemented here is the simplest possible, so each saved result is written in the same checkpoint file and therefore only loading the latest saved results is possible.

# How do I test this PR?

- For both widgets, try to run a query / mining and click on **Save** to save results. Then, re-launch the notebook and load the previous results by clicking on **Load**, and verify that the results are indeed the same.
- It was noted that we should raise a warning in case the server for searching or mining has changed since the last time we saved our results. To test that (for either widget): click on **Save**, then in the following cell run `xxx_widget.database_name = "other_db"` and/or `xxx_widget.xxx_server_version = "other version"`, and then click on **Load**. The file will be loaded anyway, but you should see a warning like the following.
<img width="400" alt="Screenshot 2020-12-03 at 18 14 24" src="https://user-images.githubusercontent.com/17013890/101063715-6bcd1880-3593-11eb-9a26-25b0dde1e20f.png">
